### PR TITLE
Add new methods to array

### PR DESCRIPTION
**Pull request by janczer**
New methods:
1) `at`
```
a = [1, "a", 10, 5]
a.at(0) #=> 1
```
2) `clear`
```
a = [1, "a", 10, 5]
a.clear
a #=> []
```

3) `concat`
```
a = [1, "a", 10, 5]
a.concat(["qwe", 1])
a #=> [1, "a", 10, 5, "qwe", 1]
```

4) `count`
```
a = [1, "a", 10, 5, 1]
a.count #=> 5

b = ["a", "b", "c", "d", "a", 2, "a"]
a.count("a") #=> 3

a = [1, 2, 3, 5, 1, 33, 4, 9, 0]
a.count do |i|
    i > 2
end
#=> 5
```

5) `rotate`
```
a = [1, "a", 10, 5, 1]
a.rotate #=> ["a", 10, 5, 1, 1]
a.rotate(3) #=> [5, 1, 1, "a", 10]
```

### CODE REVIEW

File: vm/array.go [hunk 1]
```diff
@@ -247,4 +267,144 @@ var builtinArrayMethods = []*BuiltInMethod{
...
+				}
+
+				arg := args[0]
+				findInt, findIsInt := arg.(*IntegerObject)
+				findString, findIsString := arg.(*StringObject)
+				findBoolean, findIsBoolean := arg.(*BooleanObject)
+
+				for i := 0; i < len(arr.Elements); i++ {
+					elInt, isInt := arr.Elements[i].(*IntegerObject)
+					if isInt && findIsInt && elInt.Value == findInt.Value {
```
Comment (st0012) [inline vm/array.go]: I think you should write some helper functions for these type checking and comparison operations. You can use them in other array's methods or hash's methods.

File: vm/array.go [hunk 2]
```diff
@@ -191,6 +198,19 @@ var builtinArrayMethods = []*BuiltInMethod{
 	{
 		Fn: func(receiver Object) builtinMethodBody {
 			return func(vm *VM, args []Object, blockFrame *callFrame) Object {
+				if len(args) != 0 {
```
Comment (st0012) [inline vm/array.go]: I think we can ignore arg check in `shift`, just ignore them.

File: vm/array_test.go [hunk 3]
```diff
@@ -53,6 +53,38 @@ func TestPushMethod(t *testing.T) {
 	}
 }
 
+func TestShiftMethod(t *testing.T) {
```
Comment (st0012) [inline vm/array_test.go]: I think we can separate functional tests and error tests into two different test methods.

**Comment by st0012** [comment]
BTW it seems like `at` method's sample code in PR's introduction is wrong?

**Comment by janczer** [comment]
Yes, sample code for `at` is wrong.

File: vm/array.go (hunk 1 above)
Comment (janczer) [inline vm/array.go]: I add function `equal`, but I not sure your meant it.

**Comment by st0012** [comment]
@janczer Nice job! After branch is updated I'll merge it.

**Pull request merged**